zebra-consensus/src/checkpoint/tests.rs

```diff
@@ -367,22 +366,55 @@ async fn continuous_blockchain(restart_height: Option<block::Height>) -> Result<
             }
         }
 
+        // Check that we have the correct number of verify tasks
+        if let Some(block::Height(restart_height)) = restart_height {
+            let restart_height = restart_height as usize;
+            if restart_height >= blockchain_len - 1 {
+                assert_eq!(
+                    handles.len(),
+                    0,
+                    "unexpected number of verify tasks for restart height: {:?}",
+                    restart_height,
+                );
+            } else {
+                assert_eq!(
+                    handles.len(),
+                    blockchain_len - restart_height - 2,
+                    "unexpected number of verify tasks for restart height: {:?}",
+                    restart_height,
+                );
+            }
+        } else {
+            assert_eq!(
+                handles.len(),
+                blockchain_len - 1,
+                "unexpected number of verify tasks with no restart height",
+            );
+        }
+
+        /// SPANDOC: wait on spawned verification tasks for restart height {?restart_height} {?network}
         while let Some(result) = handles.next().await {
             result??.map_err(|e| eyre!(e))?;
         }
 
         // Final checks
         assert_eq!(
             checkpoint_verifier.previous_checkpoint_height(),
-            FinalCheckpoint
+            FinalCheckpoint,
+            "unexpected previous checkpoint for restart height: {:?}",
+            restart_height,
         );
         assert_eq!(
             checkpoint_verifier.target_checkpoint_height(),
-            FinishedVerifying
+            FinishedVerifying,
+            "unexpected target checkpoint for restart height: {:?}",
+            restart_height,
         );
         assert_eq!(
             checkpoint_verifier.checkpoint_list.max_height(),
-            block::Height(9)
+            block::Height(9),
```

Comment:
If the `CONTINUOUS_MAINNET_BLOCKS` or `CONTINUOUS_TESTNET_BLOCKS` change(this can be done by just adding or removing a test vector for whatever reason) this test will fail. I think we can compare the  `checkpoint_verifier.checkpoint_list.max_height()` with the last item from `checkpoint_list` of line 260.
Maybe also a short comment on why this is `9` and not `10` which is the last checkpoint added.

Reply:
Yeah I thought about doing this, but then I forgot about it.
That would implement the rest of #1836.
(And hard-coded constants aren't great either.)

Thanks for the reminder!

Reply:
This is implemented in 15689c05.